src/Dev/Constraint/SSListContains.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace SilverStripe\Dev\Constraint;
+
+use PHPUnit_Framework_ExpectationFailedException;
+use SilverStripe\Dev\SSListExporter;
+use SilverStripe\View\ViewableData;
+
+/**
+ * Constraint for checking if a SS_List contains items matching the given
+ * key-value pairs.
+ *
+ * Class SSListContainsOnly
+ * @package SilverStripe\Dev\Constraint
+ */
+class SSListContains extends \PHPUnit_Framework_Constraint
+{
+
+    protected $matches = [];
+
+    protected $has_leftover_items = false;
```

Comment:
Can we please use lowerCamelCase (psr-2) for new protected vars, and a phpdoc would be great if you like. :)